.github/setup_evap/action.yml

```diff
@@ -0,0 +1,39 @@
+name: "Setup EvaP Testing Environment"
+description: "Sets up the nix environment with caching and copy localsettings"
+
+inputs:
+  shell:
+    description: "name of development shell to use"
+    required: false
+    default: .#evap-dev
+  start-db:
+    description: "whether or not to run the database in the background"
+    required: false
+    default: false
+  npm-ci:
+    description: "whether or not to run `npm ci`"
+    required: false
+    default: false
```

Comment:
> Could/should we make it so that the node_modules are part of a dev shell, but not of a non-dev shell?

Yes. At the moment, although we get Node and npm through nix, we don't get the node dependencies through nix. Doing so is theoretically possible, in a similar way that it works for Python: Modify `NODE_PATH` (or so) and point them to the nix store. However, as far as I can tell, no one has come up with a good way to do this yet (at least not as streamlined as poetry2nix). This is also because some npm packages assume to be placed in `node_modules` in the project root and some assume `node_modules` to be writable (at least that is what I've read). So I decided that `npm ci` is good enough for now and wanted to look at this again another time.